model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java

```diff
@@ -22,63 +22,72 @@
 import java.util.stream.Stream;
 
 import jakarta.persistence.EntityManager;
-import jakarta.persistence.Query;
 import jakarta.persistence.TypedQuery;
 import org.keycloak.connections.jpa.JpaConnectionProvider;
+import org.keycloak.models.GroupModel;
+import org.keycloak.models.GroupProvider;
 import org.keycloak.models.KeycloakSession;
 import org.keycloak.models.OrganizationModel;
 import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
 import org.keycloak.models.jpa.entities.OrganizationEntity;
 import org.keycloak.models.utils.KeycloakModelUtils;
 import org.keycloak.organization.OrganizationProvider;
 
 public class JpaOrganizationProvider implements OrganizationProvider {
 
     private final EntityManager em;
+    private final GroupProvider groupProvider;
+    private final KeycloakSession session;
 
     public JpaOrganizationProvider(KeycloakSession session) {
+        this.session = session;
         JpaConnectionProvider jpaProvider = session.getProvider(JpaConnectionProvider.class);
         this.em = jpaProvider.getEntityManager();
+        groupProvider = session.groups();
     }
 
     @Override
     public OrganizationModel createOrganization(RealmModel realm, String name) {
         throwExceptionIfRealmIsNull(realm);
         OrganizationEntity entity = new OrganizationEntity();
+        GroupModel group = createOrganizationGroup(realm, name);
 
-        entity.setId(KeycloakModelUtils.generateId());
+        entity.setId(group.getId());
```

Comment:
Yeah, I was not sure too because a 1:1 makes things a bit simpler while still making more clear the relationship between the two.

I'll change to an FK instead.